src/integer/vec_z/from.rs

```diff
@@ -0,0 +1,263 @@
+    /// less or equal to `0`.
+    ///
+    /// # Example
+    /// ```
+    /// use math::integer::VecZ;
+    /// use math::utils::VectorDirection;
+    ///
+    /// let vector = VecZ::new(5, VectorDirection::ColumnVector).unwrap();
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type
+    /// [`InvalidMatrix`](MathError::InvalidMatrix)
+    /// if the number of rows is `0`.
+    /// - Returns a [`MathError`] of type [`OutOfBounds`](MathError::OutOfBounds)
+    /// if the number of rows is negative or it does not fit into an [`i64`].
+    pub fn new(
+        num_entries: impl TryInto<i64> + Display + Copy,
+        orientation: VectorDirection,
+    ) -> Result<Self, MathError> {
+        match orientation {
+            VectorDirection::RowVector => Ok(VecZ {
+                matrix: MatZ::new(1, num_entries)?,
+            }),
+            VectorDirection::ColumnVector => Ok(VecZ {
+                matrix: MatZ::new(num_entries, 1)?,
+            }),
+        }
+    }
+}
+
+impl FromStr for VecZ {
+    type Err = MathError;
+
+    /// Creates a [`VecZ`] row-vector with entries in [`Z`] from a [`String`].
+    /// The format of that string looks like this `[1,2,3]` for a row vector
+    /// with three entries (`1` in the first row, `2` in the second one, ...)
+    ///
+    /// Parameters:
+    /// - `string`: the vector representation as a string
+    ///
+    /// Returns a [`VecZ`] or an error, if the vector is not formatted in a suitable way,
+    /// the number of rows is too big (must fit into [`i64`]), or if the regular expression
+    /// inside of the function could not be processed.
+    ///
+    /// # Example
+    /// ```
+    /// use math::integer::VecZ;
+    /// use std::str::FromStr;
+    ///
+    /// let string = String::from("[[1, 2, 3]]");
+    /// let matrix = VecZ::from_str(&string).unwrap();
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`InvalidMatrix`](MathError::InvalidMatrix)
+    /// if the vector is not formatted in a suitable way,
+    /// the number of rows is too big (must fit into [`i64`]).
+    /// - Returns a [`MathError`] of type
+    /// [`InvalidStringToCStringInput`](MathError::InvalidStringToCStringInput)
+    /// if an entry contains a Nul byte.
+    /// - Returns a [`MathError`] of type
+    /// [`InvalidStringToZInput`](MathError::InvalidStringToZInput)
+    /// if an entry is not formatted correctly.
+    fn from_str(string: &str) -> Result<Self, MathError> {
+        let entries_string = parse_matrix_string(string)?;
+        let (num_rows, num_cols) = find_matrix_dimensions(&entries_string)?;
+        let mut vector: VecZ;
+        match (num_rows, num_cols) {
+            (1, _) => vector = VecZ::new(num_cols, VectorDirection::RowVector)?,
+            (_, 1) => vector = VecZ::new(num_rows, VectorDirection::ColumnVector)?,
+            (_, _) => return Err(MathError::InvalidStringToVectorInput(String::from(
```

Comment:
should be possible to wirte it like this
```suggestion
            _ => return Err(MathError::InvalidStringToVectorInput(String::from(
```
else consider including the amount of rows and columns in the error message